crates/sequencer/src/deposit_data_mempool.rs

```diff
@@ -0,0 +1,45 @@
+use std::collections::VecDeque;
+
+use citrea_evm::system_contracts::Bridge;
+use reth_primitives::{self, address};
+use reth_rpc_types::{TransactionInput, TransactionRequest};
+
+#[derive(Clone, Debug)]
+pub struct DepositDataMempool {
+    accepted_deposit_txs: VecDeque<Vec<u8>>,
+    limit_per_block: usize,
```

Comment:
Why do we need limit per block? Add comment plz

Reply:
Given the fact that sys txs are included in the block gas limit, we should be careful to not to block other transactions + a sys tx should not fail here, so we also want to limit it to not to fail any sys tx at the beginning of the block (i.e. if you have 500 dep tx, due to gas, they may not be included, so it panics - we don't want that)

Considering the deposit amounts to be allowed, and the block count, this is convenient